extension/src/extension.ts

```diff
@@ -218,57 +216,101 @@ async function runCodeNatively(document: vscode.TextDocument) {
 
   const user32Path = getPath(["irvine", "User32.Lib"]);
 
-  //creates a new file based on the document the user is working on
-  const tempFile = document;
+  const currentDirectory = document.uri.fsPath.slice(
+    0,
+    document.uri.fsPath.length - basename(document.fileName).length
+  );
+
+  const baseFilename = basename(document.fileName).slice(
+    0,
+    -extname(document.fileName).length
+  ).replaceAll(/\s/g,"_");
+  const tempBaseFilename =
+    (basename(document.fileName).slice(0, -extname(document.fileName).length) +
+    "-temp").replaceAll(/\s/g,"_");
+  const extension = [".asm", ".obj", ".exe"];
```

Comment:
While this isn't awful it would be better to just used named constants. Not that I think you should change this.

Reply:
to me it helped with the confusion since i had contants named rmObj and i didn't want to have too many name contants
there's also a significance for the order of each item in this array as we start with the asm file we create the obj file and then we execute the exe file in that order